### PR TITLE
New recipe openmvg/2.0

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/10790

To pass v2 pipeline, this PR needs:
- conan v2 support of `ceres-solver` recipe (https://github.com/conan-io/conan-center-index/pull/14855)
- conan v2 support of `coin-clp` recipe (https://github.com/conan-io/conan-center-index/pull/15836)
- fix lz4 traits in `flann` recipe (https://github.com/conan-io/conan-center-index/pull/16355)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
In file included from /home/conan/w/prod/BuildSingleReference/.conan/data/ceres-solver/2.1.0/_/_/package/b1072642c81a88991afa656acd85c877932538ff/include/ceres/internal/autodiff.h:152:0,
                 from /home/conan/w/prod/BuildSingleReference/.conan/data/ceres-solver/2.1.0/_/_/package/b1072642c81a88991afa656acd85c877932538ff/include/ceres/autodiff_cost_function.h:130,
                 from /home/conan/w/prod/BuildSingleReference/.conan/data/ceres-solver/2.1.0/_/_/package/b1072642c81a88991afa656acd85c877932538ff/include/ceres/ceres.h:37,
                 from /home/conan/w/prod/BuildSingleReference/.conan/data/openmvg/2.0/_/_/build/64dfdd808a505f8687fad1d257cd11c4d1094a26/src/src/openMVG/multiview/rotation_averaging_l1.cpp:33:
/home/conan/w/prod/BuildSingleReference/.conan/data/ceres-solver/2.1.0/_/_/package/b1072642c81a88991afa656acd85c877932538ff/include/ceres/jet.h:194:13: error: specialization of â€˜template<class ... _Tp> struct std::common_typeâ€™ in different namespace [-fpermissive]
```

OpenMVG seems to rely on a ceres-solver header which hits a GCC bug fixed in GCC7 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480).

**Comment by SpaceIm** [comment]
Converted to draft until merge of https://github.com/conan-io/conan-center-index/pull/15836 & https://github.com/conan-io/conan-center-index/pull/16355 to allow passing v2 pipeline.

**Comment by SpaceIm** [comment]
```
ERROR: Missing binary: coin-lemon/1.3.1:8631cf963dbbb4d7a378a64a6fd1dc57558bc2fe

coin-lemon/1.3.1: WARN: Can't find a 'coin-lemon/1.3.1' package binary '8631cf963dbbb4d7a378a64a6fd1dc57558bc2fe' for the configuration:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[options]
fPIC=True
shared=False

ERROR: Missing prebuilt package for 'coin-lemon/1.3.1'
Check the available packages using 'conan list coin-lemon/1.3.1:* -r=remote'
or try to build locally from sources using the '--build=coin-lemon/1.3.1' argument

More Info at 'https://docs.conan.io/2/knowledge/faq.html#error-missing-prebuilt-package'
```

@danimtb @uilianries coud you generate v2 packages of coin-lemon please?

**Comment by SpaceIm** [comment]
Errors in v2 pipeline for Linux/all shared (https://c3i.jfrog.io/c3i/misc-v2/logs/pr/16356/13-linux-gcc/openmvg/2.0//420b80fc6515bf813bb6636821a1a154a4010d07-build.txt) likely come from a conan issue in CMakeDeps: https://github.com/conan-io/conan/issues/13560

**Comment by SpaceIm** [comment]
@uilianries @prince-chrismc @SSE4 @jwillikers could you review this one please? It's ready for 1 month, and I don't think it can pass v2 pipeline on Linux without a fix of https://github.com/conan-io/conan/issues/13560 in conan client.

File: recipes/openmvg/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        self.requires("ceres-solver/2.1.0")
+        self.requires("coin-clp/1.17.7")
+        self.requires("coin-lemon/1.3.1")
+        self.requires("coin-osi/0.108.7")
+        self.requires("coin-utils/2.11.6")
+        self.requires("eigen/3.4.0", transitive_headers=True)
+        self.requires("flann/1.9.2", transitive_headers=True, transitive_libs=True)
+        self.requires("libjpeg/9e")
+        self.requires("libpng/1.6.39")
+        self.requires("libtiff/4.4.0")
```
Comment (SpaceIm) [inline recipes/openmvg/all/conanfile.py:71]:
```suggestion
        self.requires("libtiff/4.5.0")
```

let's bump libtiff

**Comment by SpaceIm** [comment]
```
ERROR: Missing binary: ceres-solver/2.1.0:de0060c16e17cd1c56cbcceed5a819e6ab30b19b
```

@RubenRBS could you re-generate v2 packages of ceres-solver please? https://github.com/conan-io/conan-center-index/pull/16854 has changed RREV of eigen and therefore changed package id of ceres-solver (ceres-solver depends on eigen, and in conan v2 full ref of header-only libs contributes to package id of downstream).

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
Shouldn't be a blocker here, but it's worth noting that compilation fails with C++20 due to usage of https://en.cppreference.com/w/cpp/types/result_of in public headers of a dependency (`coin-lemon`):

```
[85/228] Building CXX object openMVG/sfm/CMakeFiles/openMVG_sfm.dir/pipelines/global/sfm_global_engine_relative_motions.cpp.o
FAILED: openMVG/sfm/CMakeFiles/openMVG_sfm.dir/pipelines/global/sfm_global_engine_relative_motions.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DLEMON_ONLY_TEMPLATES -DOPENMVG_STD_UNORDERED_MAP -DopenMVG_sfm_EXPORTS -I/Users/spaceim/.conan2/p/t/openm243a43d6de290/b/src/src -I/Users/spaceim/.conan2/p/t/openm243a43d6de290/b/src/src/third_party/stlplus3 -I/filesystemSimplified -isystem /Users/spaceim/.conan2/p/eigen3d88c0279cc26/p/include/eigen3 -isystem /Users/spaceim/.conan2/p/cerea4cf7b92f85513/p/include -isystem /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include -isystem /Users/spaceim/.conan2/p/flann3d294762a4bf7/p/include -isystem /Users/spaceim/.conan2/p/lz40156eeb4105bc/p/include -isystem /Users/spaceim/.conan2/p/ceres408a6b7f071eb/p/include -isystem /Users/spaceim/.conan2/p/ceres408a6b7f071eb/p/include/ceres -isystem /Users/spaceim/.conan2/p/ceres408a6b7f071eb/p/include/ceres/internal/miniglog -m64 -stdlib=libc++ -O3 -std=c++20 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -fPIC -m64 -stdlib=libc++ -MD -MT openMVG/sfm/CMakeFiles/openMVG_sfm.dir/pipelines/global/sfm_global_engine_relative_motions.cpp.o -MF openMVG/sfm/CMakeFiles/openMVG_sfm.dir/pipelines/global/sfm_global_engine_relative_motions.cpp.o.d -o openMVG/sfm/CMakeFiles/openMVG_sfm.dir/pipelines/global/sfm_global_engine_relative_motions.cpp.o -c /Users/spaceim/.conan2/p/t/openm243a43d6de290/b/src/src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp
In file included from /Users/spaceim/.conan2/p/t/openm243a43d6de290/b/src/src/openMVG/sfm/pipelines/global/sfm_global_engine_relative_motions.cpp:12:
In file included from /Users/spaceim/.conan2/p/t/openm243a43d6de290/b/src/src/openMVG/graph/graph.hpp:12:
In file included from /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/connectivity.h:22:
In file included from /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/dfs.h:26:
In file included from /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/list_graph.h:28:
In file included from /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/bits/graph_extender.h:25:
In file included from /Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/bits/default_map.h:23:
/Users/spaceim/.conan2/p/coin-f4d42f84a4725/p/include/lemon/bits/array_map.h:91:19: error: no member named 'construct' in 'std::allocator<lemon::concepts::MappableDigraphComponent<lemon::concepts::BaseGraphComponent>::Constraints<lemon::ListGraph>::Dummy>'
        allocator.construct(&(values[id]), Value());
        ~~~~~~~~~ ^
```

**Comment by SpaceIm** [comment]
> @uilianries @prince-chrismc @SSE4 @jwillikers could you review this one please? It's ready for 1 month, and I don't think it can pass v2 pipeline on Linux without a fix of [conan-io/conan#13560](https://github.com/conan-io/conan/issues/13560) in conan client.

I can try a workaround for this Linux issue (inject `VirtualRunEnv` in build scope).

**Review by prince-chrismc — APPROVED** [review]
I very very very quickly glanced at this and nothing glared at me, this will go in the progress over perfection bucket today :)

**Pull request merged**